postgres/datadog_checks/postgres/connections.py

```diff
@@ -92,8 +127,51 @@ def close_all_connections(self):
                     success = False
         return success
 
-    def _terminate_connection_unsafe(self, dbname):
-        db, _ = self._conns.pop(dbname, ConnectionWithTTL(None, None))
+    def done(self, dbname: str) -> None:
```

Comment:
The API seems a bit off to me. So applications will basically run this:

```python
conn = c.get_connection('dogs')
conn.execute(...)
c.done('dogs')
```

Which works, but requires database name context to be passed around to close the connection. I find that a bit challenging to work with and invites some hard-to-debug bugs if the wrong dbname is used.

I think the ideal API would be something like this:

```python
with c.get_connection('dogs') as conn:
    conn.execute(...)
    # the done() is implicit in the contextmanager finalizer
```

or alternatively:

```python
conn, done = c.get_connection('dogs')
try:
    conn.execute(...)
finally:
    done()
```

Reply:
that code looks a lot nicer. I will try to implement the context manager one! thank you for the suggestion

Reply:
Can this function be eliminated now with the latest changes?